src/services/directoryServices/CollectionDirectoryService.ts

```diff
@@ -85,55 +113,74 @@ class CollectionDirectoryService {
     return fileOrder
   }
 
-  async listAllCollections(sessionData) {
+  async listAllCollections(
+    sessionData: UserWithSiteSessionData
+  ): Promise<GitDirectoryItem[]> {
     const filesOrDirs = await this.baseDirectoryService.list(sessionData, {
       directoryName: "",
     })
-    return filesOrDirs.reduce((acc, curr) => {
-      if (
-        curr.type === "dir" &&
-        !ISOMER_TEMPLATE_DIRS.includes(curr.name) &&
-        curr.name.slice(0, 1) === "_"
-      )
-        acc.push({
-          name: curr.path.slice(1),
-          type: "dir",
-        })
-      return acc
-    }, [])
+    return filesOrDirs.reduce(
+      (acc: GitDirectoryItem[], curr: GitDirectoryItem) => {
+        if (
+          curr.type === "dir" &&
+          !ISOMER_TEMPLATE_DIRS.includes(curr.name) &&
+          curr.name.startsWith("_")
+        ) {
+          acc.push({
+            name: curr.name.slice(1),
+            type: "dir",
+            path: "",
+            size: 0,
+            addedTime: 0,
+          })
+        }
+        return acc
+      },
+      []
+    )
   }
 
-  async listFiles(sessionData, { collectionName }) {
+  async listFiles(
+    sessionData: UserWithSiteSessionData,
+    { collectionName }: { collectionName: string }
+  ): Promise<GitDirectoryItem[]> {
     const files = await this.collectionYmlService.listContents(sessionData, {
       collectionName,
     })
-
     return this.convertYmlToObjOrder(files)
   }
 
-  async createDirectory(sessionData, { collectionName, objArray }) {
+  async createDirectory(
+    sessionData: UserWithSiteSessionData,
+    {
+      collectionName,
+      objArray,
+    }: { collectionName: string; objArray: GitDirectoryItem[] }
+  ): Promise<{ newDirectoryName: string; items: GitDirectoryItem[] }> {
     if (ISOMER_TEMPLATE_PROTECTED_DIRS.includes(collectionName))
       throw new ConflictError(protectedFolderConflictErrorMsg(collectionName))
     if (/[^a-zA-Z0-9- ]/g.test(collectionName)) {
-      // Contains non-allowed characters
       throw new BadRequestError(
         "Special characters not allowed in collection name"
       )
     }
     const slugifiedCollectionName = slugifyCollectionName(collectionName)
     await this.collectionYmlService.create(sessionData, {
       collectionName: slugifiedCollectionName,
+      orderArray: undefined,
```

Comment:
TODO: check why we add this, also check why we are not using the `objArray` defined above